### PR TITLE
Convert to ESM

### PR DESCRIPTION
**Pull request by wmertens**

### CODE REVIEW

**Comment by wmertens** [comment]
The failing test is trying to import the esm module as cjs which should't work.

**Comment by raineorshine** [comment]
The CJS test can be removed since it will no longer be supported.

**Comment by wmertens** [comment]
All good now 😃

**Review by raineorshine — CHANGES_REQUESTED** [review]

**Comment by raineorshine** [comment]
@wmertens Just checking in to see if you have an update.

File: test/e2e.sh [hunk 1]
```diff
@@ -10,26 +9,27 @@ verdaccio_config=$temp_dir/verdaccio-config.yaml
 
 # cleanup on exit
 cleanup() {
-
+  # turn off stop on error
+  set +e
```
[the remark this answers is not in the document]
Comment (wmertens) [reply test/e2e.sh:13]: IIRC this was causing the verdaccio shutdown to fail. Not sure any more :-/

File: package.json [hunk 2]
```diff
@@ -20,20 +20,21 @@
...
     "lint:lockfile": "lockfile-lint",
-    "lint:markdown": "markdownlint \"**/*.md\" --ignore node_modules --ignore build --config .markdownlint.js",
+    "lint:markdown": "markdownlint \"**/*.md\" --ignore node_modules --ignore build --config .markdownlint.cjs",
     "lint:src": "eslint --cache --cache-location node_modules/.cache/.eslintcache --ignore-path .gitignore --report-unused-disable-directives .",
     "prepare": "src/scripts/install-hooks && test/bun-setup.sh",
     "prepublishOnly": "npm run build",
     "prettier": "prettier . --check",
     "test": "npm run test:unit && npm run test:e2e",
-    "test:unit": "mocha test test/package-managers/*",
+    "test:unit": "vitest test/package-managers/*",
```
[the remark this answers is not in the document]
Comment (wmertens) [reply package.json:37]: how did I drop it? Not sure why that message though.

[the remark this answers is not in the document]
Comment (raineorshine) [reply package.json:37]: `mocha test test/package-managers/*` specifies two paths: `./test` and `./test/package-managers/*`. This edit removes the `./test` path from the `test:unit` script (which is the main test suite for the project).

[the remark this answers is not in the document]
Comment (wmertens) [reply package.json:37]: ah maybe because it always used chdir and vitest uses workers?

[the remark this answers is not in the document]
Comment (raineorshine) [reply package.json:37]: `process.chdir` is only used in `getAllPackages.test.ts`, and it appears to be a remnant of some local testing that another developer did. The tests pass without it, so I removed it in 688c07f3a7fcda4a879ab3476fb5c70972e7505e.

**Comment by wmertens** [comment]
I'm trying to rebase but

```
npm ERR! command bash -c node install.js
npm ERR! Failed to find package "@oven/bun-linux-x64". You may have used the "--no-optional" flag when running "npm install".
npm ERR! Failed to find package "@oven/bun-linux-x64-baseline". You may have used the "--no-optional" flag when running "npm install".
```

**Comment by raineorshine** [comment]
Sorry about that! I got the same thing, but I was hoping it would work once the lock file was updated. It installs fine in ci mode. 

Try `npm install -f bun`. That worked for me. I'll circle back with a better fix soon.

**Comment by raineorshine** [comment]
Okay. I think since bun is not very cross-platform compatible, I will move it into a separate optional test script, and remove it from devDependencies.

**Comment by raineorshine** [comment]
Okay, I've completely offloaded the bun tests to a separate script.

**Comment by raineorshine** [comment]
Blocked on https://github.com/raineorshine/npm-check-updates/pull/1391#discussion_r1551595674

I'm going to close this, as it's probably too far behind to be useful now. Thank you anyway!

**Pull request closed without merging**